xla/service/algebraic_simplifier.cc

```diff
@@ -4626,11 +4626,56 @@ Status AlgebraicSimplifierVisitor::HandleBroadcast(HloInstruction* broadcast) {
   return OkStatus();
 }
 
+HloInstruction* MakeNewCompare(HloInstruction* compare, HloInstruction* a,
+                               HloInstruction* b) {
+  return compare->AddInstruction(HloInstruction::CreateCompare(
+      compare->shape(), a, b, compare->comparison_direction()));
+}
+
 Status AlgebraicSimplifierVisitor::HandleCompare(HloInstruction* compare) {
   HloInstruction* lhs;
   HloInstruction* rhs;
   CHECK(Match(compare, m::Compare(m::Op(&lhs), m::Op(&rhs))));
 
+  // Gt(Min(a,b), a) -> False
```

Comment:
This does not preserve NaNs.

Reply:
ok, removed all optimizations which return `Fase`